script/un-s4.R

```diff
@@ -0,0 +1,117 @@
+pkgload::load_all()
+
+root <- "~/git/R/r-dbi/RSQLite"
+
+if (!exists("parsed_full")) {
+  parsed_full <- parse_package(root)
+} else {
+  message("Reusing parse data from ", dirname(dirname(parsed_full$filename[[1]])))
+}
+
+parsed_nested <-
+  parsed_full %>%
+  nest(data = -filename)
+
+# If necessary, use a slice here
+parsed <-
+  parsed_nested[, ] %>%
+  unnest(data)
+
+method_idx <- map_lgl(parsed$code, ~ {
+  list <- as.list(.x)
+  if (length(list) >= 4) {
+    identical(list[[1]], rlang::sym("setMethod")) &&
+      is.call(list[[4]]) &&
+      identical(list[[4]][[1]], rlang::sym("function"))
+  } else {
+    FALSE
+  }
+})
+
+pwalk(parsed[method_idx, ], function(filename, code, srcref, parse_data) {
+  set_method_idx <- (parse_data$parent == 0)
+  stopifnot(sum(set_method_idx) == 1)
+  set_method_id <- parse_data$id[set_method_idx]
+  set_method_children_idx <- (parse_data$parent == set_method_id)
+  set_method_children_id <- parse_data$id[set_method_children_idx]
+
+  stopifnot(grepl("^function", parse_data$text[set_method_children_idx][[7]]))
```

Comment:
In parse_data$text, \(x) is not converted to function(x), so we might want to test for it too.
moreover starting with "function" is not a guarantee that we have a function, as in `function1(x)`